### DISCUSSION TITLE
Prisma queryRaw throws error when using template string variable in table name

### DISCUSSION THREAD

**Discussion by chenliu9** [prisma/prisma#12817] [Q&A] [answered]
Below are what I have tried to run,

```javascript
let result = await prisma.$queryRaw`SELECT * from "Person"` // This works fine
console.log(result)

const model = 'Person'
result = await prisma.$queryRaw`SELECT * from "${model}"` // This throws the error
```

The second **$queryRaw** throws the following error,

```shell
Invalid `prisma.queryRaw()` invocation:

    Raw query failed. Code: `42P01`. Message: `relation "$1" does not exist`
```

Below are Prisma logs from two **queryRaw** runs,

```shell
{"level":20,"time":1649915257507,"pid":6135,"hostname":"MBP","prisma":{"clientVersion":"3.11.1"},"timestamp":"2022-04-14T05:47:37.507Z","query":"SELECT * from \"Person\"","params":"[]","duration":2,"target":"quaint::connector::metrics","msg":"Query performed in 2 msec"}
  console.log
    [
      {
        id: 29,
        name: 'Rob',
        email: 'rob@example.com',
        age: 30,
        roles: null
      }
    ]

{"level":20,"time":1649915257513,"pid":6135,"hostname":"Chens-MBP","prisma":{"clientVersion":"3.11.1"},"timestamp":"2022-04-14T05:47:37.513Z","query":"SELECT * from \"$1\"","params":"[\"Person\"]","duration":1,"target":"quaint::connector::metrics","msg":"Query performed in 1 msec"}
```

As we can see, the $1 parameter is set correctly in the second queryRaw run. But for some unknown reasons, **queryRaw** throws the **relation "$1" does not exist error**.

**Comment by chenliu9** [comment]
For reference, a similar issue was reported in https://github.com/prisma/prisma/issues/5083, which got closed due to inactivity. The reported issue in https://github.com/prisma/prisma/issues/5083 was not addressed.

**Comment by austincrim** [comment, marked as answer]
Hey @chenliu9 👋 ,

We have this note [in our docs](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#using-variables) regarding `$queryRaw`:

> Note: You cannot pass a table or column name into a tagged template placeholder.

If you really need to have a dynamic table name, you would have to do plain string concatenation before passing it in to `$queryRaw`.

**Reply by chenliu9** [reply]
Got it. Thanks @austincrim!

**Reply by Uanela** [reply]
It does makes sense, I recall when people where mimicking nextjs for doing it, the answer was something like a function can be executed like this with backticks, basically $executeRaw is a function.

**Comment by fbaralle** [comment]
@chenliu9 

I made it work using `queryRawUnsafe` because it doesn't modify your template literals behind the scenes
https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryrawunsafe

But be careful with this, you might be vulnerable to sql injections

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
